### DISCUSSION TITLE
Can I use GitHub Pages to host my blog?

### DISCUSSION THREAD

**Discussion by quinnlovescode** [community/community#21561] [Other Feature Feedback, Questions, & Ideas]
<p>I’m trying to put together a new blog and I found out about github pages from a friend and I think I want to use it for my blog. Can I do this? How do I use github pages for a blog?</p>

**Comment by tzmanics** [comment, marked as answer]
<p>:wave: Great question, yes you can! I recommend it because it’s easy to get going and get your blog live. Plus, it’s free to host your static site :money_mouth_face: There are some great articles that help you get started. The GitHub Pages main site will walk you through set up by asking you questions as you scroll down: <a href="https://pages.github.com/" title="GitHub Pages" rel="nofollow noopener">https://pages.github.com/</a></p>
<p><a class="mention" href="/benbalter">@benbalter</a> also wrote a nice, succinct post on how to get started: <a href="https://github.com/blog/2289-publishing-with-github-pages-now-as-easy-as-1-2-3" title="1-2-3" rel="nofollow noopener">https://github.com/blog/2289-publishing-with-github-pages-now-as-easy-as-1-2-3</a></p>
<p>There are also really great tools like <a href="https://jekyllrb.com/" rel="nofollow noopener">Jekyll</a>and <a href="https://github.com/HubPress/hubpress.io" rel="nofollow noopener">HubPress</a> that help you build and manage your GitHub Pages site.</p>
<p>Here are some articles for these tools too:<br>
<a href="https://developer.telerik.com/featured/getting-started-with-jekyll/" rel="nofollow noopener">https://developer.telerik.com/featured/getting-started-with-jekyll/</a></p>
<p><a href="http://www.techelex.org/hubpress/" rel="nofollow noopener">http://www.techelex.org/hubpress/</a></p>
<p>Hope these help! Can’t wait to see your blog :smiley:</p>
<p>Happy Coding!</p>

**Comment by quinnlovescode** [comment]
<p><a class="mention" href="/tzmanics">@tzmanics</a> Wow! Thank you so much for all this information. this is so helpful. </p>

**Comment by nanisadw3** [comment]
Yes, GitHub Pages is one of the most popular platforms for hosting personal and technical blogs.
  
  ### Best options to get started:
  1. **Jekyll (Zero config):** GitHub automatically builds Jekyll. You can pick an existing theme, name your
repository `<username>.github.io`, and push markdown posts inside a `_posts/` directory.
  2. **Astro / Hugo (Blazing fast & modern):** Extremely fast static site generators that can be deployed
automatically to GitHub Pages via GitHub Actions (Settings -> Pages -> Source: GitHub Actions).
  3. **Custom domain & SSL:** GitHub Pages provides free custom domain support (e.g. `blog.yourname.com`) and
automatically provisions an HTTPS/SSL certificate from Let's Encrypt.
  
  It is 100% free, version-controlled with Git, and has no server maintenance overhead.
  
  If this helps clarify things for you, please consider marking this as the accepted answer!
